lib/segment.ex

```diff
@@ -1,14 +1,12 @@
 defmodule Segment do
+  require Logger
+  @service Application.get_env(:segment, :sender_impl, Segment.Analytics.Batcher)
 
-  @type status :: :ok | :error
-
-  @spec start_link(binary) :: { Segment.status, pid }
-  def start_link(write_key) do
-    Agent.start_link(fn -> write_key end, name: __MODULE__)
+  def start_link(api_key) do
+    @service.start_link(api_key)
   end
 
-  def write_key() do
-    Agent.get(__MODULE__, fn(state) -> state end)
+  def start_link(api_key, adapter) do
```

Comment:
What's this case used for?

Reply:
i was going to use it for testing. basically replacing the adapter dynamically allows you to add a mock adapter and assert the call. might not be necessary yet but i put it in